### PR TITLE
feat: migrate to the new version of Jupiter's SecurityPolicy

### PR DESCRIPTION
**Pull request by marcambier**
**Issue**

https://github.com/gravitee-io/issues/issues/7991

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.8.0-feat-jupiterPlans-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-apikey/2.8.0-feat-jupiterPlans-SNAPSHOT/gravitee-policy-apikey-2.8.0-feat-jupiterPlans-SNAPSHOT.zip)

### CODE REVIEW

File: src/main/java/io/gravitee/policy/apikey/ApiKeyPolicy.java [hunk 1]
```diff
@@ -65,17 +65,14 @@ public String id() {
...
-        optApiKey.ifPresent(apiKey -> ctx.setInternalAttribute(ATTR_INTERNAL_API_KEY, apiKey));
-
-        return Single.just(optApiKey.isPresent());
+    public Maybe<SecurityToken> extractSecurityToken(RequestExecutionContext ctx) {
+        final Optional<String> apiKey = extractApiKey(ctx);
+        if (apiKey.isPresent()) {
+            ctx.setInternalAttribute(ATTR_INTERNAL_API_KEY, apiKey.get());
+            return Maybe.just(SecurityToken.forApiKey(apiKey.get()));
+        }
+        return Maybe.empty();
```
Comment (ytvnr) [inline src/main/java/io/gravitee/policy/apikey/ApiKeyPolicy.java:75]: I'm not sure but yu can maybe do something like:
```suggestion
        return extractApiKey(ctx)
        .map(apiKey -> {
            ctx.setInternalAttribute(ATTR_INTERNAL_API_KEY, apiKey);
            return Maybe.just(SecurityToken.forApiKey(apiKey));
        })
        .orElse(Maybe.empty());
```

File: src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java [hunk 2]
```diff
@@ -218,4 +220,21 @@ private Subscription fakeSubscriptionFromCache(boolean isExpired) {
         }
         return subscription;
     }
+
+    protected void assertUnauthorizedResponseBody(String responseBody) {
+        assertThat(responseBody).isEqualTo("Unauthorized");
+    }
+
+    protected OngoingStubbing<Optional<Subscription>> whenSubscriptionSearch(ApiKey apiKey) {
```
Comment (ytvnr) [inline src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java:228]:
```suggestion
    protected OngoingStubbing<Optional<Subscription>> whenSearchingSubscription(ApiKey apiKey) {
```

File: src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java [hunk 3]
```diff
@@ -218,4 +220,21 @@ private Subscription fakeSubscriptionFromCache(boolean isExpired) {
...
+        return when(
+            getBean(SubscriptionService.class)
+                .getByApiAndSecurityToken(eq(apiKey.getApi()), securityTokenMatches(apiKey.getKey()), eq(apiKey.getPlan()))
+        );
+    }
+
+    private SecurityToken securityTokenMatches(String apiKeyValue) {
```
Comment (ytvnr) [inline src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java:235]:
```suggestion
    private SecurityToken securityTokenMatcher(String apiKeyValue) {
```

**Review by ytvnr — COMMENTED** [review]
some suggestions but not blocking

File: src/main/java/io/gravitee/policy/apikey/ApiKeyPolicy.java (hunk 1 above)
Comment (marcambier) [reply src/main/java/io/gravitee/policy/apikey/ApiKeyPolicy.java:75]: I think you're right.
Is that just for style ? (I find my way more readable, but just a personal opinion).

File: src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java (hunk 2 above)
Comment (marcambier) [reply src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java]: suggestion applied :+1: Thanks

File: src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java (hunk 3 above)
Comment (marcambier) [reply src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java]: suggestion applied :+1: Thanks

**Review by jhaeyaert — APPROVED** [review]
👏

**Pull request merged**